scripts/aio_cleanup.py

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python
+import datetime
+import dateutil.parser
+import os
+import requests
+
+from novaclient import client
+
+JENKINS_API_USER = os.environ['JENKINS_API_USER']
+JENKINS_API_PASS = os.environ['JENKINS_API_PASS']
+
+USERNAME = os.environ['OS_USERNAME'],
+API_KEY = os.environ['OS_PASSWORD'],
+TENANT_ID = os.environ['OS_TENANT_NAME'],
+AUTH_URL = os.environ['OS_AUTH_URL'],
+
+
+def jenkins_node(name, slave_names):
+    return name in slave_names
+
+
+def main():
+    nova = client.Client(2, USERNAME, API_KEY, TENANT_ID, AUTH_URL,
+                         region_name="IAD")
+
+    url = "http://jenkins.openstack-ansible.com/computer/api/json"
+    resp = requests.get(url, auth=(JENKINS_API_USER, JENKINS_API_PASS))
+    slaves = resp.json()['computer']
+
+    slave_names = [slave['displayName'] for slave in slaves]
+    print("Jenkins Slaves: %s" % slave_names)
+
+    slaves = 0
+    servers = nova.servers.list()
+    for server in servers:
+        error = server.status == 'ERROR'
+        is_slave = jenkins_node(server.name, slave_names)
+
+        created = dateutil.parser.parse(server.created)
+        created_local_tz = datetime.datetime.now(created.tzinfo)
+        six_hrs_ago = created_local_tz - datetime.timedelta(hours=6)
+        coming_of_age = created_local_tz - datetime.timedelta(minutes=7)
+
+        adult = created < coming_of_age
+        old = created < six_hrs_ago
+        if is_slave:
+            slaves += 1
+
+        msg = "Instance:{0}; Slave:{1}; Status:{2}; Adult:{3}; Old:{4}"
+        print(msg.format(server.name, is_slave, server.status, adult, old))
+
+        if (server.name.startswith('Jenkins-OM-AIO')
```

Comment:
It would be helpful if this were a parameter or environment variable so we can specify if we want to delete a specific slave without having to delete all of them. I was thinking we could run this script as the last part of the pipeline job, though I guess that won't work if the script can only delete servers that have been created six hours ago too.